test/extended/testdata/build-secrets/s2i-binary-dir/.s2i/bin/run

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Ensure none of the build config inject secrets still exist in the file system
+for s in $(ls /tmp/secret? secret?); do
+    if [ -s "${s}" ]; then
+        echo "Found secret file which should have been removed: ${s}"
+        exit 1
```

Comment:
You're using TestContainerOutput to check the output of the pod, and the doc for TestContainerOutput implies that it checks that the output contains your argument, not that it matches it exactly.  That's why i'm asking where you're testing for this.  I don't think your current test would fail if the secret files were still present.
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L2047

Is there somewhere else that i'm missing where you're doing an exact match against the output or otherwise ensuring "Found secret" is not in the output?